pytuya/__init__.py

```diff
@@ -259,35 +250,39 @@ def status(self):
         return result
 
     def set_status(self, on, switch=1):
-        """on is a bool
+        """
+        Set status of the device to ON or OFF.
+        
+        Args:
+            on(bool):  True for ON, False for OFF.
+            switch(int): The switch to set
         """
         # open device, send request, then close connection
-        command = ON if on else OFF
         if isinstance(switch, int):
             switch = str(switch)  # index and payload is a string
-        payload = self.generate_payload(command, dps_id=switch)
-        #print('payload %r' % payload)
+        payload = self.generate_payload('set', {switch:on})
 
         data = self._send_receive(payload)
         log.debug('set_status received data=%r', data)
 
         return data
 
     def set_timer(self, num_secs):
-        """num_secs should be an integer
         """
-        # FIXME / TODO support schemas? Accept timer id number as parameter?
-
+        Set a timer.
+        
+        Args:
+            num_secs(int): Number of seconds
+        """
         # Dumb heuristic; Query status, pick last device id as that is probably the timer
         status = self.status()
         devices = status['dps']
         devices_numbers = list(devices.keys())
         devices_numbers.sort()
         dps_id = devices_numbers[-1]
-        command = SET
-        payload = self.generate_payload(command, dps_id=dps_id, value=num_secs)
+
+        payload = self.generate_payload('set', {dps_id:num_secs})
```

Comment:
I'm not keen on this style, its easy to make typos in the string - that's why I used constants in the original code. Using constants means that a typo has a chance of showing up as a NameError.

Reply:
What do you think about combining the two approaches? We could still use constants but use a dict  for passing the dps value. Would that be a solution for you or don't you like that also?

Reply:
What do you think about combining the two approaches? We could still use constants but use a dict  for passing the dps value. Would that be a solution for you or don't you like that also?

Reply:
I missed the dict being passed! Yes that would be good.

Reply:
Do you want me to split that and create another pull request, so that you can incorporate the data to be passed as dict?